delegation/src/Pos/Delegation/Logic/VAR.hs

```diff
@@ -482,12 +483,12 @@ dlgRollbackBlocks dlgBlunds = do
         let proxySKs = getDlgPayload payload
             issuers = map pskIssuerPk proxySKs
             backDeleted = issuers \\ map pskIssuerPk duPsks
-            edgeActions = map (DlgEdgeDel . addressHash) backDeleted
-                       <> map DlgEdgeAdd duPsks
-        transCorrections <- calculateTransCorrections $ HS.fromList edgeActions
-        let pskOp = SomeBatchOp (map GS.PskFromEdgeAction edgeActions) <> transCorrections
+            edgeActions =
+                map (DlgEdgeDel . addressHash) backDeleted <> map DlgEdgeAdd duPsks
+        transCorrections <- calculateTransCorrections $ edgeActions
+        let pskOp = SomeBatchOp (map GS.PskFromEdgeAction $ edgeActions) <> transCorrections
         -- we should also delete issuers from "posted this epoch already"
-        let postedOp = SomeBatchOp $ map (GS.DelPostedThisEpoch . addressHash) issuers
+        let postedOp = SomeBatchOp $ map (GS.DelPostedThisEpoch . addressHash) $ toList issuers
         pure $ pskOp <> postedOp
```

Comment:
Tbh, I don't understand what is goings ons in this file. Did you forget to revert smth here?
AFAIU, some changes are not necessary and not useful, can you revert them please?

Reply:
So apparently all changes in this file are either cosmetic or optimization to avoid extra conversion between `HashSet` and list. Putting them into a separate commit would make review easier.

Reply:
They are indeed.